### DISCUSSION TITLE
CrystalSymmetry of trigonal system, rhombohedral to the hexagonal conversion in MTEX

### DISCUSSION THREAD

**Discussion by WenxiLee** [mtex-toolbox/mtex#1968] [Ask Anything]
Hi all,

We have an orientation map of corundum (Al2O3) indexed with a rhombohedral unit cell convention. However, I can only find a crystal symmetry definition for a hexagonal unit cell in MTEX, as shown in the documentation (https://mtex-toolbox.github.io/SlipSystemAlumina.html). I'm wondering if there's a way to define crystal symmetry with a rhombohedral unit cell in MTEX to plot an inverse pole figure. Alternatively, is there a function available in MTEX to convert from the rhombohedral (hP) unit cell convention to the hexagonal (hR) convention so that I can work with the hexagonal unit cell crystalSymmetry in my analysis?

![image](https://github.com/mtex-toolbox/mtex/assets/98835657/b37c07b0-9a83-403b-809d-bdfd43731bd6)

**Comment by ralfHielscher** [comment]
Hi Wenxi Li,

using MTEX you can define whatever unit cell you want. The command [`crystalSymmetry`](https://mtex-toolbox.github.io/crystalSymmetry.crystalSymmetry.html) allows to specify length and angles of the a, b and c axes.

Using the command `slipSystem` you can then define any slip system you want.

Ralf.

**Reply by WenxiLee** [reply]
Hi Ralf, 

Thank you for your response. I attempted to convert the orientation indexed using the rhombohedral cell to the hexagonal (hR) convention. However, I'm still encountering inconsistency in the IPF (Inverse Pole Figure) coloring. Could you please assist me by reviewing my code?I appreciate your help in advance.
(1)
CS1=crystalSymmetry('SpaceId',167,[4.759,4.759,12.993],[90*degree,90*degree,120*degree],'X||a','Z||c');
rot = rotation.byEuler(euler_data(gg,1)*degree,euler_data(gg,2)*degree,euler_data(gg,3)*degree); % my data are Euler angles
matrix=[];matrix=rot.matrix';  %inverse opration, crystal to sample frame
trans_mtx=[];trans_mtx=[1 -1 0;0 1 -1;1 1 1];  %HP->HR operation
rot_matrix=[];rot_matrix=trans_mtx*matrix;
ori=[];ori=orientation.byMatrix(rot_matrix,CS1);
colors=[];colors=oM.orientation2color(ori);
(2)  I attempted to specify rhombohedral cell crystal symmetries within MTEX, but the resulting crystal symmetry plot appears unusual.
CS2=crystalSymmetry('SpaceId',167,[5.128 5.128 5.128],[55.27 55.27 55.27]*degree);
![image](https://github.com/mtex-toolbox/mtex/assets/98835657/6dda5008-5276-4e73-bbb9-caa824e233e3)
